File: Python/KthLargestElementInArray.py
def findKthLargest(nums,k):

    def partition(nums,left,right):
        fill = left
        pivot = nums[right]

        for i in range(left,right):
            if nums[i] < pivot:
                temp = nums[fill]
                nums[fill] = nums[i]
                nums[i] = temp
                fill += 1

        temp = nums[fill]
        nums[fill] = nums[right]
        nums[right] = temp

        return fill


    left = 0
    right = len(nums)-1
    k = len(nums) - k

    while left < right:
        pivotIndex = partition(nums,left,right)

        if pivotIndex < k:
            left = pivotIndex + 1
        elif pivotIndex > k:
            right = pivotIndex-1
        else:
            break


    return nums[k]

File: Javascript/KthLargestElementInArray.js
var findKthLargest = function (nums,k){


    function partition(nums,left,right){
        let fill = left
        let pivot = nums[right]

        for(let i =left;i<right;i++){
            if(nums[i]<pivot){
                let temp = nums[fill]
                nums[fill] = nums[i]
                nums[i] = temp
                fill += 1
            }
        }

        let temp = nums[fill]
        nums[fill] = nums[right]
        nums[right] = temp

        return fill
    }

    let left = 0
    let right = nums.length-1

    k = nums.length-k

    while(left<right){
        let pivotIndex = partition(nums,left,right)

        if(pivotIndex < k){
            left = pivotIndex + 1
        }
        else if (pivotIndex > k){
            right = pivotIndex + 1
        }
        else{
            break
        }
    }

    return nums[k]
}